src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.ActionStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final ActionStrategy operationStrategy;
+
+    public ShopServiceImpl(ActionStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            ActionHandler handler = operationStrategy.getHandler(transaction.getAction());
+            handler.apply(transaction.getFruit(), transaction.getValue());
+        }
```

Comment:
The `process` method should ensure that the result balance after performing transactions is positive. According to the checklist, you need to check the result balance before saving it in the Storage and throw a RuntimeException if the balance is negative.